src/app/encoder.cpp

```diff
@@ -28,16 +28,17 @@
 using namespace chip;
 using namespace chip::System;
 
-static uint16_t doEncodeApsFrame(BufBound & buf, ClusterId clusterId, EndpointId sourceEndpoint, EndpointId destinationEndpoint,
-                                 EmberApsOption options, GroupId groupId, uint8_t sequence, uint8_t radius, bool isMeasuring)
+static uint16_t doEncodeApsFrame(Encoding::LittleEndian::BufferWriter & buf, ClusterId clusterId, EndpointId sourceEndpoint,
+                                 EndpointId destinationEndpoint, EmberApsOption options, GroupId groupId, uint8_t sequence,
+                                 uint8_t radius, bool isMeasuring)
 {
 
     uint8_t control_byte = 0;
     buf.Put(control_byte) // Put in a control byte
         .Put16(clusterId)
         .Put8(sourceEndpoint)
         .Put8(destinationEndpoint)
-        .Put(options, sizeof(EmberApsOption))
+        .EndianPut(options, sizeof(EmberApsOption))
```

Comment:
Maybe as a future todo we should ensure that we never use EndianPut.
Spec generally specifies sizes in bytes and the typedef of EmberApsOption does not have a say in how big our payloads should be.